plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/JwtTokenManager.java

```diff
@@ -54,50 +60,49 @@
     public String createToken(Authentication authentication) {
         return createToken(authentication.getName());
     }
-    
+
     /**
      * Create token.
      *
      * @param userName auth info
      * @return token
      */
     public String createToken(String userName) {
-        
+
         long now = System.currentTimeMillis();
-        
+
         Date validity;
-        
+
         validity = new Date(now + nacosAuthConfig.getTokenValidityInSeconds() * 1000L);
-        
+
         Claims claims = Jwts.claims().setSubject(userName);
         return Jwts.builder().setClaims(claims).setExpiration(validity)
                 .signWith(Keys.hmacShaKeyFor(nacosAuthConfig.getSecretKeyBytes()), SignatureAlgorithm.HS256).compact();
     }
-    
+
     /**
      * Get auth Info.
      *
      * @param token token
      * @return auth info
      */
     public Authentication getAuthentication(String token) {
-        Claims claims = Jwts.parserBuilder().setSigningKey(nacosAuthConfig.getSecretKeyBytes()).build()
-                .parseClaimsJws(token).getBody();
-        
+        Claims claims = jwtParser.parseClaimsJws(token).getBody();
+
         List<GrantedAuthority> authorities = AuthorityUtils
                 .commaSeparatedStringToAuthorityList((String) claims.get(AUTHORITIES_KEY));
-        
+
         User principal = new User(claims.getSubject(), "", authorities);
         return new UsernamePasswordAuthenticationToken(principal, "", authorities);
     }
-    
+
     /**
      * validate token.
      *
      * @param token token
      */
     public void validateToken(String token) {
-        Jwts.parserBuilder().setSigningKey(nacosAuthConfig.getSecretKeyBytes()).build().parseClaimsJws(token);
+        jwtParser.parseClaimsJws(token);
```

Comment:
Are there thread safety issues with using singletons?

Reply:
i will test the thread safety issues